crates/matrix-sdk-base/src/client.rs

```diff
@@ -338,6 +338,16 @@ impl BaseClient {
                                             .or_default()
                                             .insert(member.sender().to_owned(), member.into());
                                     }
+
+                                    if *member.membership() == MembershipState::Invite {
+                                        // Remove any profile previously stored for the invited
```

Comment:
Perhaps it would make sense to explain why we're doing this as well.

Reply:
Good idea, i'll add a small function helper so as to avoid the repetition of code and comments then.